src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -3,12 +3,63 @@
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
+import core.basesyntax.utility.RegistrationException;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final Integer MIN_AGE = 18;
+    private static final Integer MIN_SYMBOLS = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User should not be null");
+        }
+        validateUserLogin(user);
+        validateUserAge(user);
+        validateUserPassword(user);
+        if (storageDao.get(user.getLogin()) == null) {
+            storageDao.add(user);
```

Comment:
If a user with the same login already exists in the storage, the method should throw an exception instead of adding a new user.

Reply:
Done